crates/oct-cloud/src/aws.rs

```diff
@@ -348,14 +422,36 @@ impl Resource for Ec2Instance {
         self.public_ip = instance.public_ip_address.clone();
         self.public_dns = instance.public_dns_name.clone();
 
+        // Save instance info to the state file
+        let mut state = State::load("state.json");
+        state.add_instance(InstanceState {
+            id: self.id.clone().unwrap_or_else(|| "".to_string()),
+            arn: self.arn.clone().unwrap_or_else(|| "".to_string()),
+            public_ip: self.public_ip.clone().unwrap_or_else(|| "".to_string()),
+            public_dns: self.public_dns.clone().unwrap_or_else(|| "".to_string()),
+            region: self.region.clone(),
+            ami: self.ami.clone(),
+            instance_type: self.instance_type.clone().into(),
+            name: self.name.clone(),
+        });
+        state.save("state.json");
+
         Ok(())
     }
 
     async fn destroy(&mut self) -> Result<(), Box<dyn std::error::Error>> {
+        let mut state = State::load("state.json");
+
+        let instance_id = self.id.clone().ok_or("No instance id")?;
+
         self.client
             .terminate_instance(self.id.clone().ok_or("No instance id")?)
             .await?;
 
+        // Remove instance info from the state file
+        state.remove_instance(&instance_id);
+        state.save("state.json");
+
```

Comment:
You already do it in CLI. What is the expected place to communicate with the state?